modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -1127,6 +1122,32 @@
     [(str \$ aggr-name) {aggr-group aggr-name}]
     extracted-aggr))
 
+(defn- adjust-distinct-aggregations
+  "This function transforms `aggr-expr'` as in [[expand-aggregations]] so identifiers representing array that is
+  a set of _distinct_ values are wrapped in `{$size...}.
+
+  `aggr-expr` is expected to be a clause that is a result of [[extract-aggregations]]. For details see its docstring.
+
+  Distinct values are computed using the `$addToSet` in a `$group` stage. `$size` transforms them to actual count.
+
+  If there is a need for more _pre-post_ transformations (as per results of [[expand-aggregations]]), this function
+  should be generalized."
```

Comment:
It's not clear what this means. I'd either turn this into a `TODO` comment that calls out a specific weakness to be fixed later, or just remove this sentence.

Reply:
By `pre-post transformations` I mean transformations that are necessary on `:post` key of `expand-aggregations` return value. Even if meaning of the comment was evident, its added value is questionable. Removed.